File: app/src/main/java/com/smarthane/admiral/component/app/AppLifecyclesImpl.java
package com.smarthane.admiral.component.app;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;

import com.smarthane.admiral.core.base.AppComponent;
import com.smarthane.admiral.core.base.delegate.AppLifecycles;
import com.smarthane.admiralcomponent.BuildConfig;
import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;

/**
 * @author smarthane
 * @time 2019/11/10 11:16
 * @describe
 */
public class AppLifecyclesImpl implements AppLifecycles {

    @Override
    public void attachBaseContext(@NonNull Context base) {

    }

    @Override
    public void onCreate(@NonNull Application application) {
        if (LeakCanary.isInAnalyzerProcess(application)) {
            // This process is dedicated to LeakCanary for heap analysis.
            // You should not init your app in this process.
            return;
        }
        // leakCanary内存泄露检查
        AppComponent
                .get()
                .fetchExtras()
                .put(RefWatcher.class.getName(), BuildConfig.USE_CANARY ? LeakCanary.install(application) : RefWatcher.DISABLED);
    }

    @Override
    public void onTerminate(@NonNull Application application) {

    }
}

File: module_jetpack/src/main/java/com/smarthane/admiralcomponent/jetpack/app/AppLifecyclesImpl.kt
package com.smarthane.admiralcomponent.jetpack.app

import android.app.Application
import android.content.Context
import com.smarthane.admiral.core.base.delegate.AppLifecycles

/**
 * @author smarthane
 * @time 2019/11/10 18:16
 * @describe
 */
class AppLifecyclesImpl : AppLifecycles {
    override fun attachBaseContext(base: Context) {
    }

    override fun onCreate(application: Application) {
    }

    override fun onTerminate(application: Application) {
    }
}